packages/block-library/src/image/edit.native.js

```diff
@@ -801,6 +803,11 @@ export class ImageEdit extends Component {
 												? additionalImageProps
 												: {} ) }
 										/>
+										{ isGif && (
```

Comment:
The `GIF` badge displays as expected, but gets covered up by the `Featured` badge if I set the GIF to be a featured image, with parts of the underlying `GIF` badge remaining visible:

https://user-images.githubusercontent.com/2998162/155327535-01f9f54d-2c75-4414-977b-c434d9b551b6.mov

My expectation would be for the badge to be completely replaced by the `Featured` badge if I set an image as featured. 

The `Featured` badge was implemented via [a Badge component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/badge/README.md) and is found on this line in the code:

https://github.com/WordPress/gutenberg/blob/320b89d038b8f976ce06ae5233d23eae7c19d377/packages/block-library/src/image/edit.native.js#L743

I think it could make sense to create new `badgeLabel` and `badgeDisplay` variables that determine the values for the Badge component's `label` and `show` props. This component has already been "signed off" from a design point of view and could be iterated on in future, with this approach. What do you think? Is this an approach you think would make sense?

_As an aside: I'd originally implemented the Badge component in the way I did based on the code in this approved (but not merged) PR that was created for this same "`GIF` badge" issue, which adds to my thinking about this being a good approach: https://github.com/WordPress/gutenberg/pull/19111_

Reply:
I've updated the code to fix the badge issue; however, still working on the other issue about not using the URL to verify if the image is an animated GIF or not because I'm having some issues with `ArrayBuffer` conversions @hypest.

See below video for the badge fix:

https://user-images.githubusercontent.com/13263478/155558234-b35210f5-b94e-49cc-a1c8-f357d63cde65.mp4

Reply:
I wonder if we should actually display both the Featured and the GIF badges at the same time. I wouldn't consider that a blocker for this PR though.

From a design POV @iamthomasbishop , what do you think about displaying both "Featured" and "GIF" when that's the case for the image

Reply:
@hypest @SiobhyB  I think it could work if we stack the badges vertically (Featured badge first thenthe secondary GIF badge below it, both aligned along their left edges). Stacking them horizontally would probably be more likely to cause issues, especially for small images.

Reply:
Thanks for the follow-up Thomas! For this, [the Badge component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/badge/README.md) would need to be refactored to accept more than one argument, which might not be so simple. Perhaps separating this out into a separate issue would be the best option here, unless this is deemed a blocker from a design POV or @illusaen has different thoughts?

Reply:
Ah, okay, thank you for clarifying! If that's the case and we can (for now) only show one badge, I would agree with your earlier suggestion:

> My expectation would be for the badge to be completely replaced by the Featured badge if I set an image as featured.